pkg/query/selection.go

```diff
@@ -157,8 +158,23 @@ func mergeCriteria(c1 []Criterion, c2 []Criterion) ([]Criterion, error) {
 	}
 
 	for _, newCriterion := range c2 {
+
+		// disallow duplicate label queries
+		if newCriterion.Type == LabelQuery {
```

Comment:
this should be by key IMO.
`key1 = val1|key1 = val2` => `key1=val1 or key1=val2` which is equal to `key1 in [val1,val2]`

Reply:
Ok - initially i thought we could allow key1 = val1|key1 = val2 because it works but if you prefer we can disallow - I've added a second commit disallowing dup keys

Reply:
on a second thought `key1 = val1|key1 = val2` should actually mean that key1 is equal to both val1 and val2 at the same time (since values are an array, it should be mean that values contains both val1 and val2).

What do you think?